docs/reference/anatomy-of-a-recipe.rst

```diff
@@ -0,0 +1,329 @@
+.. _anatomy-of-a-recipe:
+
+Anatomy of a recipe
+===================
+
+This page provides a break down of the keys in complete Snapcraft recipes, as
+declared in ``snapcraft.yaml`` files.
+
+Two recipes will be examined. The first is a simple recipe from a Python-based
+snap called `yt-dlp <https://github.com/yt-dlp/yt-dlp>`_. The second is more
+complex and builds the `wethr <https://github.com/twobucks/wethr>`_ app, which
+is written in Node.
+
+Each key in the yt-dlp recipe will be covered, and the more advanced keys in
+wethr will be examined.
+
+
+Simple recipe
+-------------
+
+yt-dlp is a command line tool for extracting online videos, and is a
+self-contained Python project.
+
+.. collapse:: yt-dlp recipe
+
+  .. literalinclude:: code/yt-dlp-recipe.yaml
+    :language: yaml
+    :lines: 2-
+
+
+Top-level directives
+~~~~~~~~~~~~~~~~~~~~
+
+Several `top-level keys
+<https://snapcraft.io/docs/snapcraft-yaml-schema#p-21225-top-level-directives>`_
+define and describe the snap.
+
+
+Metadata
+^^^^^^^^
+
+For convenience, the first set of keys comprise the snap's metadata, or
+manifest. These declare information about the snap itself, such as the snap's
+name, version, summary, and description. These identifying keys allow users to
+find the snap through the Snap Store.
+
+The ``name`` key defines the name of the snap. It must start with an ASCII
+character and can only use ASCII lowercase letters, numbers, and hyphens. It
+must be between 1 and 40 characters in length. It must be unique in the Snap
+Store.
+
+The ``version`` key defines the version of the snap to the user. The maximum
+length is 32 characters.
+
+The ``summary`` key describes the software in brief. It can't exceed 79
+characters. It accepts multi-line YAML syntax.
+
+The ``description`` key describes the software in full. It also accepts
+multi-line syntax. There is no practical limit on the length of this key.
```

Comment:
Last week I learned the description has a 4,096 character limit.  I just added this to the spreadsheet.

Reply:
1df1efa25